src/commands/createFunction.ts

```diff
@@ -104,29 +118,53 @@ export async function createFunction(
 
     const localAppSettings: LocalAppSettings = new LocalAppSettings(ui, azureAccount, functionAppPath);
 
-    const templatePicks: PickWithData<Template>[] = (await templateData.getTemplates()).map((t: Template) => new PickWithData<Template>(t, t.name));
+    const languageType: string = await projectUtils.getProjectType(functionAppPath);
+
+    const templatePicks: PickWithData<Template>[] = (await templateData.getTemplates(languageType)).map((t: Template) => new PickWithData<Template>(t, t.name));
     const templatePlaceHolder: string = localize('azFunc.selectFuncTemplate', 'Select a function template');
     const template: Template = (await ui.showQuickPick<Template>(templatePicks, templatePlaceHolder)).data;
 
     if (template.bindingType !== 'httpTrigger') {
         await localAppSettings.validateAzureWebJobsStorage();
     }
 
-    const name: string = await promptForFunctionName(ui, functionAppPath, template);
+    const name: string = await promptForFunctionName(ui, functionAppPath, template, languageType);
+    const javaFuntionProperties: string[] = [];
 
     for (const settingName of template.userPromptedSettings) {
         const setting: ConfigSetting | undefined = await templateData.getSetting(template.bindingType, settingName);
         if (setting) {
             const defaultValue: string | undefined = template.getSetting(settingName);
             const settingValue: string | undefined = await promptForSetting(ui, localAppSettings, setting, defaultValue);
-
-            template.setSetting(settingName, settingValue);
+            if (languageType === TemplateLanguage.Java) {
+                javaFuntionProperties.push(`"-D${settingName}=${settingValue}"`);
+            } else {
+                template.setSetting(settingName, settingValue);
+            }
         }
     }
 
-    const functionPath: string = path.join(functionAppPath, name);
-    await template.writeTemplateFiles(functionPath);
+    let newFilePath: string;
+    if (languageType === TemplateLanguage.Java) {
+        const packageName: string = await promptForPackageName(ui);
```

Comment:
Let's prompt for package name before prompting for settings

Reply:
Sure